src/Illuminate/Collections/Collection.php

```diff
@@ -372,6 +373,35 @@ public function filter(?callable $callback = null)
         return new static(array_filter($this->items));
     }
 
+    /**
+     * Run a map and filter over each of the items.
+     *
+     * @template TMapValue
+     *
+     * @param  callable(TValue, TKey): TMapValue  $callback
+     * @param  (callable(TValue, TKey): bool)|null  $reject
+     * @return static<TKey, TMapValue>
+     */
+    public function filterMap($callable, $filter = null)
+    {
+        $filter = $filter === null
+            ? fn ($value) => (bool) $value
+            : match ((new ReflectionFunction($filter))->getNumberOfParameters()) {
```

Comment:
This reflection seems unnecessary

Reply:
PHP's built-in methods throw an exception when you pass too many arguments to them.

We check the number of arguments here to ensure that we don't hit that exception if the function only accepts a single argument (as we pass the value and the key to the filter method).

We take a similar approach in the `Collection:map` and `Arr::map` functions:

https://github.com/laravel/framework/blob/0e9053da9d709c544200260cc0be5e223ea8ff93/src/Illuminate/Collections/Arr.php#L600-L610

However, I benchmarked this approach with my approach and found that using reflection was faster than allowing the exception to be thrown and the impact of using reflection is negligible.

Reply:
Ah yes, the good ol php inbuilt functions that always destroy all your hopes 😅  I forgot about those

Reflection is indeed better than try, catch, because try catch means you need to call the function which if it had some side effects could be very undesirable